### PR TITLE
migrate pkg/controller/volume/pvc_protection_controller.go to structured logs

### PR DESCRIPTION
**Pull request by iakisme**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

xref:
https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by wangkai1994** [comment]
/sig storage
/assign humblec
/assign msau42

some fix, Thanks!

File: pkg/controller/volume/pvcprotection/pvc_protection_controller.go [hunk 1]
```diff
@@ -153,7 +153,7 @@ func (c *Controller) processNextWorkItem() bool {
 }
 
 func (c *Controller) processPVC(pvcNamespace, pvcName string) error {
-	klog.V(4).Infof("Processing PVC %s/%s", pvcNamespace, pvcName)
+	klog.V(4).InfoS("Processing PVC", pvcNamespace, "/", pvcName)
```
Comment (msau42) [inline pkg/controller/volume/pvcprotection/pvc_protection_controller.go:156]: Should this be changed to use KRef or KObj? ditto everywhere

Comment (wangkai1994) [reply pkg/controller/volume/pvcprotection/pvc_protection_controller.go:156]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging

image

I just follow the doc and fix.

Comment (msau42) [reply pkg/controller/volume/pvcprotection/pvc_protection_controller.go:156]: As part of the user guide at https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#use-klogkobj-and-klogkref-for-kubernetes-objects, it also talks about using KObj and KRef

Comment (wangkai1994) [reply pkg/controller/volume/pvcprotection/pvc_protection_controller.go]: Thanks, i know what you mean~

Comment (wangkai1994) [reply pkg/controller/volume/pvcprotection/pvc_protection_controller.go]: > As part of the user guide at https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#use-klogkobj-and-klogkref-for-kubernetes-objects, it also talks about using KObj and KRef

done~

**Comment by msau42** [comment]
/lgtm
/approve

Thanks!

**Comment by serathius** [comment]
Hey there were some errors in how klog.InfoS was used. For example
```
klog.V(2).InfoS("Keeping PVC", "PVC", klog.KRef(pvcNamespace, pvcName), " because it is still being used")
```
/cc @ehashman 

I will try to improve migration guide and add static analysis to prevent against this. 
It's not a Note to reviewers, please ask for including comparison of log output as proposed in https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#verify-log-output

**Pull request merged**